### PR TITLE
Add BatchAdd documentation to DG

### LINKED ISSUE #96: Add Batch Add feature to Developer Guide

### PR DESCRIPTION
**Pull request by prit3010**
Fixes #96 

UML Diagram missing. Will work on it

### CODE REVIEW

**Comment by chao890** [comment]
Can clean up on punctuation, otherwise LGTM.

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -243,6 +243,44 @@ _{more aspects and alternatives to be added}_
 
 _{Explain here how the data archiving feature will be implemented}_
 
+
+### BatchAdd 
+This feature is created for users to add multiple entries at once. 
+In the case of this application, there are two main reasons why our User (HR Executive) would use this.
+1. User is new and needs to import all the current data into the database.
+2. There is a new recruitment cycle and company has recruited a large number of employees
```
Comment (Kok-je) [inline docs/DeveloperGuide.md:251]: Missing a fullstop here.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -243,6 +243,44 @@ _{more aspects and alternatives to be added}_
...
+
+Moving on to the implementation, some things to note.
+- As of now, our feature only accommodates adding from a CSV file.
+- Fields does not allow for commas inside. 
+These are possible things to work on for future iterations
```
Comment (Kok-je) [inline docs/DeveloperGuide.md:256]: Missing punctuation here

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -243,6 +243,44 @@ _{more aspects and alternatives to be added}_
...
+into a `List<AddCommand>`. If file does not exist in the folder, a `FileNotFound` exception is thrown too.
+
+Step 3. Once `getInfo` returns a `List<AddCommand>`, the list will then be iterated through to execute each `AddCommand`
+If there is any duplicate Person found, the function call will be aborted and the database will be reverted to its original state.
+
+Step 4. storage#saveDatabase is then called on the current `database`, updates the database to contain the new persons added.
+
+####Design Considerations
+- Alternative 1 (Current Choice): Make use of the execution of the `AddCommand`
+  - Pros: Makes use of the Error Handling that the `AddCommand` has
```
Comment (Kok-je) [inline docs/DeveloperGuide.md:279]: Consider rephrasing, "Makes use of `AddCommand` Error Handling functionality.

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -243,6 +243,44 @@ _{more aspects and alternatives to be added}_
...
+  - Cons: `BatchAdd` will fail if Add fails
```
Comment (Kok-je) [inline docs/DeveloperGuide.md:280]: Consider rephrasing, "A failure with `Add` will cause `BatchAdd` to fail.

**Review by Kok-je — COMMENTED** [review]
Some minor punctuation and phrasing errors, otherwise was good effort in documenting the BatchAdd method. Keep up the good work Prittam!

**Review by kevinchangjk — APPROVED** [review]
LGTM.

**Pull request merged**